plugins/org.jboss.tools.openshift.egit.core/src/org/jboss/tools/openshift/egit/core/EGitUtils.java

```diff
@@ -887,15 +937,29 @@ public static String getCurrentBranch(Repository repository)
 		return branch;
 	}
 
+	/**
+	 * Returns the current branch for the given project.
+	 * 
+	 * @param project
+	 * @return
+	 * @throws CoreException
+	 * 
+	 * @see #getRepository(IProject)
+	 */
 	public static String getCurrentBranch(IProject project)
 			throws CoreException {
 		if (project == null) {
 			return null;
 		}
 		Repository repo = EGitUtils.getRepository(project);
```

Comment:
had a similar approach initially with the older syntax where closing would happen in a finally block. But seemed less readable to me. "Try with resource" +1